cmd/cmds.go

```diff
@@ -0,0 +1,109 @@
+package fscmds
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	cmds "github.com/ipfs/go-ipfs-cmds"
+	"github.com/multiformats/go-multiaddr"
+)
+
+// CmdsParameterSet defines an argument's description
+// and the set of keys to retrieve it from various sources.
+type CmdsParameterSet struct {
+	Name, Description, Environment string
+}
+
+const (
+	// ServiceName defines a default name servers and clients may use
+	// to refer to the service via namespace oriented APIs.
+	// Effectively the service root.
+	ServiceName = "fs"
+	// ServerName defines a default name servers and clients may use
+	// to form or find connections to the named server instance.
+	// (E.g. a Unix socket of path `.../$ServiceName/$ServerName`.)
+	ServerName = "server"
+)
+
+var (
+	ServiceMaddrParameter = CmdsParameterSet{
+		Name:        "api",
+		Description: "File system service multiaddr to use.",
+		Environment: "FS_API_MADDR",
+	}
+	AutoExitParameter = CmdsParameterSet{
+		Name:        "stop-after",
+		Description: `Time interval (e.g. "30s") to check if the service is active and exit if not.`,
+		Environment: "FS_STOP_AFTER",
+	}
+
+	Options = []cmds.Option{
+		// TODO: consider file-relevant cmds pkg options (symlinks, hidden attribute, etc.)
+		// for dealing with fs/mtab-like input file
+		cmds.OptionEncodingType,
+		cmds.OptionTimeout,
+		cmds.OptionStreamChannels,
+		cmds.BoolOption(cmds.OptLongHelp, "Show the full command help text."),
+		cmds.BoolOption(cmds.OptShortHelp, "Show a short version of the command help text."),
+		cmds.StringOption(ServiceMaddrParameter.Name, ServiceMaddrParameter.Description),
+		cmds.StringOption(AutoExitParameter.Name, AutoExitParameter.Description),
+	}
```

Comment:
The options used by cmds roots shouldn't be a mutable var.
This needs to be changed into a generator function like `RootOptions`.

Reply:
Yeah okay: https://github.com/djdv/go-filesystem-utils/pull/2/commits/b33706e865e97a8513f0ff39f4646d2afb9b5d3f